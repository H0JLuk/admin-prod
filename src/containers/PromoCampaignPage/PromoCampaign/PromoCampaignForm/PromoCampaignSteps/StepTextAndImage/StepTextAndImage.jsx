import React, { useCallback } from 'react';
import Template from './Templates/Template';
import PROMO_CAMPAIGNS from '../../../../../../constants/promoCampaigns';

import styles from './StepTextAndImage.module.css';

const StepTextAndImage = ({
    typePromoCampaign,
    addChangedImg,
    banners,
    texts,
    setFields,
    isCopy,
}) => {
    const onRemoveImg = useCallback((name) => {
        setFields([{ name, value: [] }]);
        addChangedImg(name[1]);
    }, [setFields, addChangedImg]);

    return (
        <div className={ styles.containerStep }>
            <div>
                <div className={ styles.title }>{ PROMO_CAMPAIGNS[typePromoCampaign].label }</div>
                <div className={ styles.container }>
                    <Template
                        banners={ banners }
                        texts={ texts }
                        isCopy={ isCopy }
                        onRemoveImg={ onRemoveImg }
                        type={ typePromoCampaign }
                    />
                </div>
            </div>
        </div>
    );
};

export default StepTextAndImage;
