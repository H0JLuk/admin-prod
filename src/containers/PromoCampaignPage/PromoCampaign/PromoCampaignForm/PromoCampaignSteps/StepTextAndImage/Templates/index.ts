import Template from './Template';

export type { TemplateProps } from './Template';

export default Template;
