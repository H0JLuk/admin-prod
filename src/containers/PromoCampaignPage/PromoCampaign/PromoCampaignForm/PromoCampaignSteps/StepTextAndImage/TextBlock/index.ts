import TextBlock from './TextBlock';

export type { TextBlockProps } from './TextBlock';

export default TextBlock;
