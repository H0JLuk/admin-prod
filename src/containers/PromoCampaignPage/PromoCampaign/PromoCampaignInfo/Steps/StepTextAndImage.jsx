import React from 'react';
import Template from './Templates/Template';

import styles from '../PromoCampaignInfo.module.css';

const TITLE_BY_TYPE = {
    NORMAL: 'Экскурсия',
    PRESENT: 'Подарок',
};

const StepTextAndImage = ({ banners, texts, type }) => (
    <div className={ styles.containerStep }>
        <div className={ styles.infoDetail }>
            <div className={ styles.title }>
                { TITLE_BY_TYPE[type] }
            </div>
            <div className={ styles.container }>
                <Template
                    banners={ banners }
                    texts={ texts }
                    type={ type }
                />
            </div>
        </div>
    </div>
);

export default StepTextAndImage;
