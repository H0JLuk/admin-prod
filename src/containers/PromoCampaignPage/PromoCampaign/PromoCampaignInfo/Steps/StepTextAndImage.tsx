import React from 'react';
import Template from './Templates/Template';
import { PromoCampaignDto } from '@types';

import styles from '../PromoCampaignInfo.module.css';

type IStepTextAndImageProps = Pick<PromoCampaignDto, 'banners' | 'type' | 'texts' >;

const TITLE_BY_TYPE = {
    NORMAL: 'Экскурсия',
    PRESENT: 'Подарок',
};

const StepTextAndImage: React.FC<IStepTextAndImageProps> = ({ banners, texts, type }) => (
    <div className={styles.containerStep}>
        <div className={styles.infoDetail}>
            <div className={styles.title}>
                {TITLE_BY_TYPE[type as keyof typeof TITLE_BY_TYPE]}
            </div>
            <div className={styles.container}>
                <Template
                    banners={banners}
                    texts={texts}
                    type={type}
                />
            </div>
        </div>
    </div>
);

export default StepTextAndImage;
