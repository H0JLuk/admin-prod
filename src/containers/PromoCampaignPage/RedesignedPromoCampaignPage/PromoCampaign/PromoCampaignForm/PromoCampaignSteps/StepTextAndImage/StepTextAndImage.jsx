import React, { useCallback } from 'react';
import Template from './Templates/Template';
import { Form } from 'antd';
import { steps } from '../../PromoCampaignFormConstants';
import PROMO_CAMPAIGNS from '../../../../../../../constants/promoCampaigns';
import styles from './StepTextAndImage.module.css';

const templateTypes = {
    excursion: 'excursion',
    gift: 'gift',
};

const StepTextAndImage = ({ state, handlerNextStep, validStepChange, addChangedImg }) => {
    const [form] = Form.useForm();
    const { typePromoCampaign } = state;

    const onFinish = useCallback((val) => {
        validStepChange(steps.visibility);
        handlerNextStep(Object.keys(val).reduce((result, key) => {
            const [fieldType, fieldName] = key.split('.');
            const value = val[key];

            return {
                ...result,
                [fieldType]: {
                    ...result[fieldType],
                    [fieldName]: value
                }
            };
        }, {}));
    }, [handlerNextStep, validStepChange]);

    const onRemoveImg = useCallback((name) => form.setFieldsValue({ [name]: [] }), [form]);

    /** @type {import('antd/lib/form').FormProps['onFieldsChange']} */
    const isChanged = useCallback(changedFields => {
        if (changedFields.some(({ touched }) => touched)) {
            validStepChange(steps.textAndBanners, false);
        }

        if (changedFields.length === 1) {
            const [fieldName] = changedFields[0].name;
            const [, imgTypeName] = String(fieldName).split('.');
            addChangedImg(imgTypeName);
        }
    }, [validStepChange, addChangedImg]);

    return (
        <Form
            id="info"
            form={ form }
            className={ styles.containerStep }
            onFinish={ onFinish }
            onFieldsChange={ isChanged }
        >
            <div className={ styles.containerStep }>
                {typePromoCampaign === PROMO_CAMPAIGNS.NORMAL.value && (
                    <div className={ styles.infoDetail }>
                        <div className={ styles.title }>{ PROMO_CAMPAIGNS.NORMAL.label }</div>
                        <div className={ styles.container }>
                            <Template
                                banners={ state.promoCampaignBanners }
                                texts={ state.promoCampaignTexts }
                                onRemoveImg={ onRemoveImg }
                                type= { templateTypes.excursion }
                            />
                        </div>
                    </div>
                )}
                {typePromoCampaign === PROMO_CAMPAIGNS.PRESENT.value && (
                    <div className={ styles.infoDetail }>
                        <div className={ styles.title }>{ PROMO_CAMPAIGNS.PRESENT.label }</div>
                        <div className={ styles.container }>
                            <Template
                                banners={ state.promoCampaignBanners }
                                texts={ state.promoCampaignTexts }
                                onRemoveImg={ onRemoveImg }
                                type= { templateTypes.gift }
                            />
                        </div>
                    </div>
                )}
            </div>
        </Form>
    );
};

export default StepTextAndImage;
