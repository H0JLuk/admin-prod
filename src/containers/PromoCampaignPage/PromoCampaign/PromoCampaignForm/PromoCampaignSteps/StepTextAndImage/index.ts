import StepTextAndImage from './StepTextAndImage';

export type { StepTextAndImageProps } from './StepTextAndImage';

export default StepTextAndImage;
