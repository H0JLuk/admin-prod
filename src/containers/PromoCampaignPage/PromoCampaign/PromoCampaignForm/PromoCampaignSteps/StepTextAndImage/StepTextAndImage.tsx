import React, { useCallback } from 'react';
import Template from './Templates/Template';
import PROMO_CAMPAIGNS from '@constants/promoCampaigns';
import { FormInstance, FormItemProps } from 'antd';
import { BannerCreateDto, BannerCreateTextDto } from '@types';
import { INFO_ROWS_KEYS } from './Templates/templateConstants';

import styles from './StepTextAndImage.module.css';

export type StepTextAndImageProps = {
    typePromoCampaign: INFO_ROWS_KEYS;
    addChangedImg: (name: FormItemProps['name']) => void;
    banners: BannerCreateDto;
    texts: BannerCreateTextDto;
    setFields: FormInstance['setFields'];
    isCopy: boolean | undefined;
};

const StepTextAndImage: React.FC<StepTextAndImageProps> = ({
    typePromoCampaign,
    addChangedImg,
    banners,
    texts,
    setFields,
    isCopy,
}) => {
    const onRemoveImg = useCallback((name) => {
        setFields([{ name, value: [] }]);
        addChangedImg(name[1]);
    }, [setFields, addChangedImg]);

    return (
        <div className={styles.containerStep}>
            <div>
                <div className={styles.title}>{PROMO_CAMPAIGNS[typePromoCampaign as INFO_ROWS_KEYS].label}</div>
                <div className={styles.container}>
                    <Template
                        banners={banners}
                        texts={texts}
                        isCopy={isCopy}
                        onRemoveImg={onRemoveImg}
                        type={typePromoCampaign}
                    />
                </div>
            </div>
        </div>
    );
};

export default StepTextAndImage;
