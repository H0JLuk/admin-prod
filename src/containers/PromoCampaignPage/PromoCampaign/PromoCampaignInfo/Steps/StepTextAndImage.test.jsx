import React from 'react';
import StepTextAndImage from './StepTextAndImage';
import { shallow } from 'enzyme';
import { promoCampaignBannerArray, promoCampaignTextsArray } from '../../../../../../__tests__/constants';

describe('<StepTextAndImage /> test', () => {
    const props = {
        banners: promoCampaignBannerArray,
        texts: promoCampaignTextsArray,
        type: 'NORMAL',
    };
    const StepTextAndImageItem = shallow(<StepTextAndImage { ...props } />);

    it('StepTextAndImage snapshot', () => {
        expect(StepTextAndImageItem.html).toMatchSnapshot();
    });

    it('normal type test', () => {
        expect(StepTextAndImageItem.find('.title').text()).toEqual('Экскурсия');
        expect(StepTextAndImageItem.find('Template').prop('type')).toEqual(props.type);
    });

    it('present type test', () => {
        const newProps = {
            ...props,
            type: 'PRESENT',
        };
        StepTextAndImageItem.setProps(newProps);
        expect(StepTextAndImageItem.find('.title').text()).toEqual('Подарок');
        expect(StepTextAndImageItem.find('Template').prop('type')).toEqual(newProps.type);
    });
});

