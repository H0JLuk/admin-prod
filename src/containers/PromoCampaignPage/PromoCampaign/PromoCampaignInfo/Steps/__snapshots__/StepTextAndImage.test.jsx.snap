// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepTextAndImage /> test StepTextAndImage snapshot 1`] = `[Function]`;
